native/src/main/java/io/ballerina/stdlib/sql/nativeimpl/OutParameterProcessor.java

```diff
@@ -162,4 +177,330 @@ public static Object get(
             return ErrorGenerator.getSQLDatabaseError(sqlException, "Error when parsing out parameter.");
         }
     }
+
+    private static Object convertToArray(String objectTypeName, Object[] dataArray, Type ballerinaType)
```

Comment:
This method is only used in this class. So I implemented here. Do we have to move this?

Reply:
We use AbstractParameterProcessor because we want the code to be flexible. If we are adding any new DB connectors later on and they want to change it they can simply ovveride it without changing it here. Its better to keep it there